pyClient/test_commands/test_merkle_tree_contract.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2015-2019 Clearmatics Technologies Ltd
+#
+# SPDX-License-Identifier: LGPL-3.0+
+
+from zeth.constants import ZETH_MERKLE_TREE_DEPTH
+from zeth.merkle_tree import MerkleTree
+from zeth.utils import extend_32bytes
+from typing import List, Any
+import test_commands.mock as mock
+
+
+TEST_VALUES = [
+    extend_32bytes(bytes.fromhex("f0")),
+    extend_32bytes(bytes.fromhex("f1")),
+    extend_32bytes(bytes.fromhex("f2")),
+    extend_32bytes(bytes.fromhex("f3")),
+    extend_32bytes(bytes.fromhex("f4")),
+    extend_32bytes(bytes.fromhex("f5")),
+    extend_32bytes(bytes.fromhex("f6")),
+    extend_32bytes(bytes.fromhex("f7")),
+    extend_32bytes(bytes.fromhex("f8")),
+    extend_32bytes(bytes.fromhex("f9")),
+    extend_32bytes(bytes.fromhex("fa")),
+    extend_32bytes(bytes.fromhex("fb")),
+    extend_32bytes(bytes.fromhex("fc")),
+    extend_32bytes(bytes.fromhex("fd")),
+    extend_32bytes(bytes.fromhex("fe")),
+    extend_32bytes(bytes.fromhex("ff")),
+]
+
+
+def assert_root(expect_root: bytes, nodes: List[bytes], msg: str) -> None:
+    if nodes[0] != expect_root:
+        print(f"FAILED: {msg}")
+        print(f"Expected: {expect_root.hex()}")
+        print("Actual  :")
+        for layer_idx in range(0, ZETH_MERKLE_TREE_DEPTH + 1):
+            layer_size = pow(2, layer_idx)
+            layer_start = layer_size - 1
+            layer = nodes[layer_start:layer_start + layer_size]
+            layer_hex = [node.hex() for node in layer]
+            print(f" {layer_hex}")
+        raise Exception(f"failed")
+
+
+# def assert_root(expect_root: bytes, root: bytes, msg: str) -> None:
+#     if root != expect_root:
+#         print(f"FAILED: {msg}")
+#         print(f"EXPECTED: {expect_root.hex()}")
+#         print(f"ACTUAL  : {root.hex()}")
+#         raise Exception(f"failed")
```

Comment:
Did you keep this comment on purpose @dtebbs or should it be removed?

Reply:
Removed.  Thanks.